FirebaseAdmin/FirebaseAdmin/Messaging/InstanceIdClient.cs

```diff
@@ -0,0 +1,181 @@
+    /// <returns>A task that completes with a <see cref="TopicManagementResponse"/>, giving details about the topic subscription operations.</returns>
+    public async Task<TopicManagementResponse> SubscribeToTopicAsync(string topic, List<string> registrationTokens)
+    {
+        try
+        {
+            return await this.SendInstanceIdRequest(topic, registrationTokens, IidSubscriberPath).ConfigureAwait(false);
+        }
+        catch (HttpRequestException e)
+        {
+            throw this.CreateExceptionFromResponse(e);
+        }
+        catch (IOException)
+        {
+            throw new FirebaseMessagingException(ErrorCode.Internal, "Error while calling IID backend service");
+        }
+    }
+
+    /// <summary>
+    /// Unsubscribes a list of registration tokens from a topic.
+    /// </summary>
+    /// <param name="topic">The topic name to unsubscribe from. /topics/ will be prepended to the topic name provided if absent.</param>
+    /// <param name="registrationTokens">A list of registration tokens to unsubscribe.</param>
+    /// <returns>A task that completes with a <see cref="TopicManagementResponse"/>, giving details about the topic unsubscription operations.</returns>
+    public async Task<TopicManagementResponse> UnsubscribeFromTopicAsync(string topic, List<string> registrationTokens)
+    {
+        try
+        {
+            return await this.SendInstanceIdRequest(topic, registrationTokens, IidUnsubscribePath).ConfigureAwait(false);
+        }
+        catch (HttpRequestException e)
+        {
+            throw this.CreateExceptionFromResponse(e);
+        }
+        catch (IOException)
+        {
+            throw new FirebaseMessagingException(ErrorCode.Internal, "Error while calling IID backend service");
+        }
+    }
+
+    /// <summary>
+    /// Dispose the HttpClient.
+    /// </summary>
+    public void Dispose()
+    {
+        this.httpClient.Dispose();
+    }
+
+    private async Task<TopicManagementResponse> SendInstanceIdRequest(string topic, List<string> registrationTokens, string path)
+    {
+        string url = $"{IidHost}/{path}";
+        var body = new InstanceIdServiceRequest
+        {
+            Topic = this.GetPrefixedTopic(topic),
+            RegistrationTokens = registrationTokens,
+        };
+
+        var request = new HttpRequestMessage()
+        {
+            Method = HttpMethod.Post,
+            RequestUri = new Uri(url),
+            Content = NewtonsoftJsonSerializer.Instance.CreateJsonHttpContent(body),
+        };
+
+        request.Headers.Add("access_token_auth", "true");
+
+        try
+        {
+            var response = await this.httpClient.SendAsync(request, default(CancellationToken)).ConfigureAwait(false);
+            var json = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            this.errorHandler.ThrowIfError(response, json);
+            return JsonConvert.DeserializeObject<TopicManagementResponse>(json);
```

Comment:
This doesn't look right. You probably need a new `InstanceIfServiceResponse` type, to which the response can be deserialized. Then you need to create a `TopicManagementResponse` off of that.